tests/test_routes.py

```diff
@@ -178,6 +182,217 @@ def test_get_customer_by_email(self):
         for record in data:
             self.assertEqual(record["email"], customers[0].email)
 
+    def test_get_customer_by_street(self):
+        """It should Get all the Customers by street"""
+
+        customers = CustomerFactory.create_batch(2)
+        addresses_to_add = AddressFactory.create_batch(4)
+
+        # Create the addresses corresponding to the customers
+        for idx, customer in enumerate(customers):
+            customer.create()
+            for addr_idx in range(2):
+                addr = addresses_to_add[idx * 2 + addr_idx]
+                resp = self.client.post(
+                    f"{BASE_URL}/{customer.id}/addresses",
+                    json=addr.serialize())
+                self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+
+        street_to_test = customers[0].addresses[0].street
+        customers_list = []
+
+        # Test if the number of returned customers is okay
+        for customer in customers:
+            for address in customer.addresses:
+                if address.street == street_to_test:
+                    customers_list.append(customer)
+                    break
+
+        resp = self.client.get(
+            BASE_URL, query_string=f"street={street_to_test}")
+
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        data = resp.get_json()
+
+        self.assertEqual(len(data), len(list(set(customers_list))))
+        count = 0
+
+        for record in data:
```

Comment:
What's this second count assertion for? Can you please add a comment like line 204 to indicate what are we checking here.

Reply:
Oh right first, we just check if the len of the customers matching the description and the len of the response are equal. This is a rudimentary check. 

The second count is a more stringent check, checking if the customers returned have at least one address satisfying the given constraints. Since we are returning all the addresses for an eligible customer, I just needed to check if at least one address satisfied the constraints.